From 2c67e5a200b775d83391fee192b93fbf0e22e96c Mon Sep 17 00:00:00 2001
From: chalesyu <574249312@qq.com>
Date: Fri, 7 Nov 2025 00:00:00 +0800
Subject: [PATCH 2/2] Fix for GX ethernet port

Upstream linux merged this patch, but u-boot not yet.
without this ethernet won't work in upstream linux.
https://lore.kernel.org/all/20250425192009.1439508-1-da@libre.computer/
---
 drivers/net/mdio_mux_meson_gxl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/mdio_mux_meson_gxl.c b/drivers/net/mdio_mux_meson_gxl.c
index 8ef3ae598b7..375948048b8 100644
--- a/drivers/net/mdio_mux_meson_gxl.c
+++ b/drivers/net/mdio_mux_meson_gxl.c
@@ -66,7 +66,7 @@ static int meson_gxl_enable_internal_mdio(struct mdio_mux_meson_gxl_priv *priv)
 	 * The only constraint is that it must match the one in
 	 * drivers/net/phy/meson-gxl.c to properly match the PHY.
 	 */
-	writel(FIELD_PREP(REG2_PHYID, EPHY_GXL_ID),
+	writel( BIT(28) | FIELD_PREP(REG2_PHYID, EPHY_GXL_ID),
 	       priv->regs + ETH_REG2);
 
 	/* Enable the internal phy */
-- 

